demisto_sdk/commands/update_release_notes/update_rn.py

```diff
@@ -0,0 +1,216 @@
+"""
+This script is used to create a release notes template
+"""
+
+import errno
+import json
+import os
+import sys
+
+from demisto_sdk.commands.common.constants import PACKS_PACK_META_FILE_NAME
+from demisto_sdk.commands.common.hook_validations.structure import \
+    StructureValidator
+from demisto_sdk.commands.common.tools import (LOG_COLORS, get_json,
+                                               pack_name_to_path, print_color,
+                                               print_error, print_warning)
+
+
+class UpdateRN:
+    def __init__(self, pack: str, update_type: str, pack_files: set, pre_release: bool = False):
+
+        self.pack = pack
+        self.update_type = update_type
+        self.pack_meta_file = PACKS_PACK_META_FILE_NAME
+        self.pack_path = pack_name_to_path(self.pack)
+        self.metadata_path = os.path.join(self.pack_path, 'pack_metadata.json')
+        self.pack_files = pack_files
+        self.pre_release = pre_release
+
+    DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+    def execute_update(self):
+        try:
+            new_version = self.bump_version_number(self.pre_release)
+        except ValueError as e:
+            print_error(e)
+            sys.exit(0)
+        rn_path = self.return_release_notes_path(new_version)
+        self.check_rn_dir(rn_path)
+        changed_files = {}
+        for packfile in self.pack_files:
+            fn, ft = self.ident_changed_file_type(packfile)
+            changed_files[fn] = ft
+        rn_string = self.build_rn_template(changed_files)
+        self.create_markdown(rn_path, rn_string)
+
+    def _does_pack_metadata_exist(self):
+        """Check if pack_metadata.json exists"""
+        if not os.path.isfile(self.metadata_path):
+            print_error('"{}" file does not exist, create one in the root of the pack'.format(
+                self.metadata_path))
+            return False
+
+        return True
+
+    def return_release_notes_path(self, input_version: str):
+        _new_version = input_version.replace('.', '_')
+        new_version = _new_version.replace('_prerelease', '')
+        return os.path.join(self.pack_path, 'ReleaseNotes', '{}.md'.format(new_version))
+
+    @staticmethod
+    def get_display_name(file_path):
+        struct = StructureValidator(file_path=file_path)
+        file_data = struct.load_data_from_file()
+        if 'name' in file_data:
+            name = file_data.get('name', None)
+        elif 'TypeName' in file_data:
+            name = file_data.get('TypeName', None)
+        else:
+            name = os.path.basename(file_path)
+            print_error(f"Could not find name in {file_path}")
+            # sys.exit(1)
+        return name
+
+    @staticmethod
+    def find_corresponding_yml(file_path):
+        if file_path.endswith('.py'):
```

Comment:
question, should this be relevant to other files in the pack? why just py/yml?

Reply:
Py files don't have a file structure where I can extract the display name so I know that if we get a python file in the git diff, it's either a script or an integration which should have the display name in the yml. The code above is to load the yml file path so we can pull the display name.